File: src/main/java/com/ocunha/domain/restaurant/filters/DistanceFilterDefinition.java
package com.ocunha.domain.restaurant.filters;

import com.ocunha.domain.restaurant.model.RestaurantSearchParams;
import com.ocunha.domain.restaurant.model.model.Restaurant;
import org.springframework.stereotype.Component;

@Component
public class DistanceFilterDefinition implements FilterDefinition {

    @Override
    public boolean filter(Restaurant restaurant, RestaurantSearchParams searchParams) {
        return isLessOrEquals(restaurant.distance(), searchParams.getDistance());
    }

}

File: src/main/kotlin/com/ocunha/domain/filter/DistanceFilterDefinition.kt
package com.ocunha.domain.filter

import com.ocunha.domain.restaurant.model.Restaurant
import com.ocunha.domain.restaurant.model.RestaurantSearchParams
import org.springframework.stereotype.Component

@Component
class DistanceFilterDefinition : FilterDefinition {
    override fun filter(restaurant: Restaurant?, searchParams: RestaurantSearchParams?): Boolean {
        return isLessOrEquals(restaurant?.distance, searchParams?.distance)
    }

}